src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -226,16 +220,31 @@ private void refreshSecretFromCloud(String arn, String versionStage) {
         try {
             GetSecretValueResponse response = secretClient.getSecret(request);
             logger.atDebug().kv("secret", arn).kv("versionStage", versionStage).log("Downloaded secret from cloud");
-            if (localStoreMap.updateWithSecret(response, getSecretConfiguration())) {
-                // Reload cache with every secret update
-                reloadCache();
-            }
-        } catch (SecretCryptoException | SecretManagerException  e) {
+            localStoreMap.updateWithSecret(response, getSecretConfiguration());
+            /*
+            Always update the in-memory cache as saving to disk may fail due to a slow TPM but that should not fail
+            a get secret IPC request
```

Comment:
We could also throw an exception with a message that the customer should retry secret retrievals in such cases. 

Previously, we designed our solution such that the local store and the cache would always be in sync with each other and reflect the same data. So, can we add a comment here or a log based on the boolean status of `updateWithSecret` to indicate whether the local store is updated or not if it isn't covered already?